demos/smart_classroom_demo/cpp_gapi/include/initialize.hpp

```diff
@@ -0,0 +1,379 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <opencv2/gapi/infer/ie.hpp>
+#include <opencv2/gapi/infer/parsers.hpp>
+
+#include "custom_kernels.hpp"
+#include "actions.hpp"
+#include "kernel_packages.hpp"
+
+namespace nets {
+    G_API_NET(FaceDetector,        <cv::GMat(cv::GMat)>, "face-detector");
+    G_API_NET(LandmarksDetector,   <cv::GMat(cv::GMat)>, "landmarks-detector");
+    G_API_NET(FaceReidentificator, <cv::GMat(cv::GMat)>, "face-reidentificator");
+    using PAInfo = std::tuple<cv::GMat, cv::GMat, cv::GMat, cv::GMat, cv::GMat, cv::GMat, cv::GMat>;
+    G_API_NET(PersonDetActionRec,  <PAInfo(cv::GMat)>,   "person-detection-action-recognition");
+} // namespace nets
+
+namespace config {
+inline char separator() {
+#ifdef _WIN32
+    return '\\';
+#else
+    return '/';
+#endif
+}
+
+bool fileExists(const std::string& name) {
+    std::ifstream f(name.c_str());
+    return f.good();
+}
+
+std::string folderName(const std::string& path) {
+    size_t found_pos;
+    found_pos = path.find_last_of(separator());
+    if (found_pos != std::string::npos)
+        return path.substr(0, found_pos);
+    return std::string(".") + separator();
+}
+
+std::vector<std::string> parseActionLabels(const std::string& in_str) {
+    std::vector<std::string> labels;
+    std::string label;
+    std::istringstream stream_to_split(in_str);
+    while (std::getline(stream_to_split, label, ',')) {
+        labels.push_back(label);
+    }
+    return labels;
+}
+
+FaceRecognizerConfig getRecConfig(const bool greedy_reid_matching, const double t_reid) {
+    FaceRecognizerConfig rec_config;
+    rec_config.reid_threshold = t_reid;
+    rec_config.greedy_reid_matching = greedy_reid_matching;
+    return rec_config;
+}
+
+bool isNetForSixActions(const std::string& model_path) {
+    CV_Assert(!model_path.empty());
+    return model_path.at(model_path.size() - 5) == '6';
```

Comment:
Could you describe in readme that the demo relies on model file name

Reply:
Your note doesn't reflect this part of code. You don't care about starting of file name like `person-detection-action-recognition-000` you care about the ending.
I suggest

```
> **NOTE**: The demo deduces a number of actions for a person detection action recognition model by its name. The demo expects the name to end with `\d.xml`, `\d` denotes a digit.
```

Reply:
BTW here is a proposal based on this problem for gapi: `cv::gapi::ie::Params` for `cv::gapi::GNetPackage` should be able to take not xml and bin (ideally they also should be replaced with just xml), but also `CNNNetwork` generated by `ie.ReadNetwork()`.
1. This would allow demos reading all the info from `CNNNetwork` and then pass it to gapi avoiding rereading the file.
2. This would remove a need in `cfgInputReshape()` because it could be done using common way of `CNNNetwork`.